lib/model/folder_sendrecv.go

```diff
@@ -1158,7 +1157,8 @@ func (f *sendReceiveFolder) shortcutFile(file, curFile protocol.FileInfo, dbUpda
 // copierRoutine reads copierStates until the in channel closes and performs
 // the relevant copies when possible, or passes it to the puller routine.
 func (f *sendReceiveFolder) copierRoutine(in <-chan copyBlocksState, pullChan chan<- pullBlockState, out chan<- *sharedPullerState) {
-	buf := make([]byte, protocol.MinBlockSize)
+	buf := protocol.GetBuf(protocol.MinBlockSize)
+	defer protocol.PutBuf(buf)
```

Comment:
We sure buf is not captured at the point of defer?

Reply:
It's just used to transfer data from one file to another. After the routine, nothing happens with it anymore.

Reply:
My point is that there the buf too small switcheroo with a new buf, and I suspect the old buf gets put back twice, and new one is left dangling.

Reply:
Probably, I'd try it in the playground to see qhat actually happens.

Reply:
I did try before committing :)  
As far as I understand all arguments to the deferred function are evaluated when the defer statement is executed. So putting the function inside an anonymous function prevents that evaluation.